admin/src/core/services/cognito.service.ts

```diff
@@ -0,0 +1,196 @@
+   *      https://github.com/bcgov/nr-forests-access-management/wiki/OIDC-Attribute-Mapping
+   * The display username is "custom:idp_username" from token.
+   */
+  private parseToken(authToken: CognitoUserSession): any {
+    const decodedIdToken = authToken.getIdToken().decodePayload();
+    const decodedAccessToken = authToken.getAccessToken().decodePayload();
+    return {
+      id_token: decodedIdToken,
+      access_token: decodedAccessToken,
+      jwtToken: authToken
+    };
+  }
+
+  /*
+      See Aws-Amplify documenation for intgration: 
+      https://docs.amplify.aws/lib/auth/social/q/platform/js/
+      https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#identity-pool-federation
+  */
+  public async init(): Promise<any> {
+
+    this.loggedOut = this.getParameterByName("loggedout");
+    if (this.loggedOut === "true") {
+      this.initialized = false;
+      return null;
+    }
+
+    let url: string =
+      this.configService.getApiBasePath() + "/api/awsCognitoConfig";
+    this.awsCognitoConfig = await this.http
+      .get(url, { observe: "body", responseType: "json" })
+      .toPromise() as AwsCognitoConfig;
+
+    let keycloakUrl: string =
+      this.configService.getApiBasePath() + "/api/keycloakConfig";
+    this.keycloakConfig = await this.http
+      .get(keycloakUrl, { observe: "body", responseType: "json" })
+      .toPromise() as KeycloakConfig;
+
+    console.log("Using cognito config = " + JSON.stringify(this.awsCognitoConfig));
+    Amplify.configure(this.awsCognitoConfig);
+
+    if (!this.awsCognitoConfig.enabled) {
+      this.fakeUser = getFakeUser();
+      this.initialized = true;
+      return null;
+    }
+
+    return new Promise(async (resolve, reject) => {
+      if (this.loggedOut === "true") {
+        console.log("stay logout");
+        resolve(null);
+      } else {
+        Auth.currentAuthenticatedUser()
+          .then(async (_userData) => {
+            console.log("_userData", _userData);
+            if (!_userData) {
+              await this.login();
+            } else {
+              await this.refreshToken();
+            }
+            this.initialized = true;
+            resolve(null);
+          })
+          .catch(async (error) => {
+            console.log("There is no current user", error);
+            await this.login();
+            resolve(null);
+          });
+      }
+    });
+  }
+
+  /**
+   * Amplify method currentSession() will automatically refresh the accessToken and idToken
+   * if tokens are "expired" and a valid refreshToken presented.
+   *   // console.log("currentAuthToken: ", currentAuthToken)
+   *   // console.log("ID Token: ", currentAuthToken.getIdToken().getJwtToken())
+   *   // console.log("Access Token: ", currentAuthToken.getAccessToken().getJwtToken())
+   *
+   * Automatically logout if unable to get currentSession().
+   */
+  async refreshToken() {
+    try {
+      console.log("Refreshing Token...");
```

Comment:
A lot of these logging statements can be removed once debugging is done. I'd keep the one showing the user's decoded access token, but most of the rest aren't that useful.

Reply:
Removed.